pkg/autodiscovery/providers/kubecontainer.go

```diff
@@ -0,0 +1,321 @@
+		containerEntityName := containers.BuildEntityName(string(container.Runtime), containerID)
+		c, errors := utils.ExtractTemplatesFromContainerLabels(containerEntityName, container.Labels)
+
+		for _, err := range errors {
+			adErrors[containerEntityName] = map[string]struct{}{err.Error(): {}}
+			log.Errorf("Can't parse template for container %s: %s", containerID, err)
+		}
+
+		c = utils.AddContainerCollectAllConfigs(c, containerEntityName)
+
+		for idx := range c {
+			c[idx].Source = names.Container + ":" + containerEntityName
+		}
+
+		configs = append(configs, c...)
+	}
+
+	log.Debugf("Generating Configs for %d pods\n", len(k.podCache))
+	for _, pod := range k.podCache {
+		var errs []error
+		containerIdentifiers := map[string]struct{}{}
+		containerNames := map[string]struct{}{}
+		for _, podContainer := range pod.Containers {
+			container, err := k.workloadmetaStore.GetContainer(podContainer.ID)
+			if err != nil {
+				log.Debugf("Pod %q has reference to non-existing container %q", pod.Name, podContainer.ID)
+				continue
+			}
+
+			log.Debugf("Considering pod %s with container id %s\n", pod.Name, podContainer.ID)
+			adIdentifier := podContainer.Name
+			if customADID, found := utils.ExtractCheckIDFromPodAnnotations(pod.Annotations, podContainer.Name); found {
+				adIdentifier = customADID
+			}
+
+			containerEntity := containers.BuildEntityName(string(container.Runtime), container.ID)
+			c, errors := utils.ExtractTemplatesFromPodAnnotations(
+				containerEntity,
+				pod.Annotations,
+				adIdentifier,
+			)
+
+			if len(errors) > 0 {
+				for _, err := range errors {
+					log.Errorf("Can't parse template for pod %s: %s", pod.Name, err)
+					errs = append(errs, err)
+				}
+				continue
+			}
+
+			_, trackedByContainer := k.containerCache[podContainer.ID]
+			if !trackedByContainer {
+				c = utils.AddContainerCollectAllConfigs(c, containerEntity)
+			} else {
+				log.Debugf("Pod %q has container %q, however container is tracked by containerCache, skipping log config creation...", pod.Name, podContainer.ID)
+			}
+
+			containerIdentifiers[adIdentifier] = struct{}{}
+			containerNames[podContainer.Name] = struct{}{}
+
+			for idx := range c {
+				c[idx].Source = "kubelet:" + containerEntity
+			}
+
+			configs = append(configs, c...)
+		}
+
+		errs = append(errs, utils.ValidateAnnotationsMatching(
+			pod.Annotations,
+			containerIdentifiers,
+			containerNames)...)
+
+		namespacedName := pod.Namespace + "/" + pod.Name
+		for _, err := range errs {
+			if _, found := adErrors[namespacedName]; !found {
+				adErrors[namespacedName] = map[string]struct{}{err.Error(): {}}
+			} else {
+				adErrors[namespacedName][err.Error()] = struct{}{}
+			}
+		}
+	}
+
+	bldr := strings.Builder{}
+	for _, c := range configs {
+		fmt.Fprintf(&bldr, "  %s %s %s\n", c.Name, c.Source, c.Digest())
+	}
+	log.Infof("KubeContainerConfigProvider#generateConfigs generated:\n%s", bldr.String())
```

Comment:
I think `INFO` is a bit too verbose, since this gets executed up to once every second, and includes *every* config, not only the changed ones.

Reply:
Agreed, setting to `Debug` for now and will switch to `DebugFunc` if [this pr](https://github.com/DataDog/datadog-agent/pull/12876) lands